core/src/main/java/org/elasticsearch/search/suggest/completion/RegexOptions.java

```diff
@@ -35,10 +38,26 @@
  * Regular expression options for completion suggester
  */
 public class RegexOptions implements ToXContent, Writeable<RegexOptions> {
-    static final String NAME = "regex";
-    static final ParseField REGEX_OPTIONS = new ParseField(NAME);
-    static final ParseField FLAGS_VALUE = new ParseField("flags", "flags_value");
-    static final ParseField MAX_DETERMINIZED_STATES = new ParseField("max_determinized_states");
+    static final ParseField REGEX_OPTIONS = new ParseField("regex");
+    private static final ParseField FLAGS_VALUE = new ParseField("flags", "flags_value");
+    private static final ParseField MAX_DETERMINIZED_STATES = new ParseField("max_determinized_states");
+
+    private static ObjectParser<Builder, Void> PARSER = new ObjectParser<>(REGEX_OPTIONS.getPreferredName(), Builder::new);
+    static {
+        PARSER.declareInt(Builder::setMaxDeterminizedStates, MAX_DETERMINIZED_STATES);
+        PARSER.declareField((parser, builder, aVoid) -> {
+            if (parser.currentToken() == XContentParser.Token.VALUE_STRING) {
+                builder.setFlags(parser.text());
+            } else if (parser.currentToken() == XContentParser.Token.VALUE_NUMBER) {
+                builder.setFlagsValue(parser.intValue());
```

Comment:
Ah nice, this also wasn't covered before when this was part of CompletionSuggestionParser?